### PR TITLE
Update custom-templates.md

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes dotnet/templating #2533: added information about sourceName and preferNameDirectory in template.json file.

### CODE REVIEW

File: docs/core/tools/custom-templates.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ The *template.json* file is placed in a *.template.config* folder in the root di
 | `identity`        | string        | A unique name for this template. |
 | `name`            | string        | The name for the template that users should see. |
 | `shortName`       | string        | A default shorthand name for selecting the template that applies to environments where the template name is specified by the user, not selected via a GUI. For example, the short name is useful when using templates from a command prompt with CLI commands. |
+| `sourceName`       | string        | The name in the source tree to replace with the name the user specifies. The template engine will look for any occurrence of the `sourceName` mentioned in the config file and replace it in file names and file contents. The value to be replaced with can be given using the `-n` or `--name` options while running a template. If no name is specified, the current directory is used.|
+| `preferNameDirectory`       | boolean        | Indicates whether to create a directory for the template if name is specified but an output directory is not set (instead of creating the content directly in the current directory). |
```
Comment (tdykstra) [inline docs/core/tools/custom-templates.md:62]: What is the default if the option isn't specified?

**Review by tdykstra — APPROVED** [review]
Looks good, one suggestion.

File: docs/core/tools/custom-templates.md (hunk 1 above)
Comment (vlada-shubina) [reply docs/core/tools/custom-templates.md]: The default value is false, added to the doc.

File: docs/core/tools/custom-templates.md [hunk 2]
```diff
@@ -58,6 +58,8 @@ The *template.json* file is placed in a *.template.config* folder in the root di
 | `identity`        | string        | A unique name for this template. |
 | `name`            | string        | The name for the template that users should see. |
 | `shortName`       | string        | A default shorthand name for selecting the template that applies to environments where the template name is specified by the user, not selected via a GUI. For example, the short name is useful when using templates from a command prompt with CLI commands. |
+| `sourceName`       | string        | The name in the source tree to replace with the name the user specifies. The template engine will look for any occurrence of the `sourceName` mentioned in the config file and replace it in file names and file contents. The value to be replaced with can be given using the `-n` or `--name` options while running a template. If no name is specified, the current directory is used.|
```
Comment (KathleenDollard) [inline docs/core/tools/custom-templates.md:61]: If the config is template.config, I think we should specify. The might be simpler as:

> The text which will be replaced with the name the user specifies. The template engine will look for any occurrence of the `sourceName`text and replace it in file names and file contents. When running the template, the user specified the new value using the `-n` or `--name` options while running a template. If no name is specified, the current directory is used.

**Review by sayedihashimi — COMMENTED** [review]
Looks good, thanks. I left a minor question in the comments.

File: docs/core/tools/custom-templates.md (hunk 2 above)
Comment (vlada-shubina) [reply docs/core/tools/custom-templates.md:61]: @KathleenDollard I can easily adapt it. Just want to double check the following sentence:
_When running the template, the user specified the new value using the -n or --name options while running a template._ 

It has 2 similar clauses (when running the template, while running a template) should we keep just one of them?

Comment (vlada-shubina) [reply docs/core/tools/custom-templates.md:61]: @sayedihashimi the behavior is as follows:
`dotnet new mytemplate` will create the template in the current directory for any value of `preferNameDirectory`. The `sourceName` value will be replaced with name of current directory.

`dotnet new mytemplate -n myName` will create the template:
- in current directory when `preferNameDirectory = false`. The `sourceName` value will be replaced with `myName`.
- in subdirectory `myName` when `preferNameDirectory = true`. The `sourceName` value will be replaced with `myName`.

**Pull request merged**